### ISSUE TITLE
Suggestion: add 'exclude' filter

### ISSUE THREAD

**Issue by acastruc** [carina-studio/ULogViewer#43]
Amazing software, truly.  Some suggestions
1. Make the filter list accessible in the right docking bar under a "Filter" tab, alongside "Marked Logs", "Categories of timestamps", "Log analysis" and "Added log files". Then if each item has a checkbox next to it the user could "apply" them by checking box.
2. Give the ability to define a filter that excludes instead of includes. For example, I often want to exclude logs with a certain category, or message substring. Then let me <s>AND</s> define multiple so I can filter like "Show all log lines except those that contain "foo" or "bar" etc. Or show me all logs except those with category="foo" or category="bar"

**Comment by hamster620** [comment]
@acastruc Thanks for your suggestion, we wil take these into account in the future release.

**Comment by acastruc** [comment]
@hamster620 I'd be happy to take a stab at this myself, do you have any dev setup documentation? Also, Im on mac but could virtualize Win if required. Thanks

**Comment by hamster620** [comment]
@acastruc You can just clone this repositary and use ```master``` branch to develop. For IDE you can use ```Visual Studio Code```, ```Rider``` or ```Visual Studio (on Windows)```. ```macOS``` is fine to develop and run ULogViewer.

**Comment by hamster620** [comment]
For core log filtering function, you can refer to [DisplayableLogFilter.cs](https://github.com/carina-studio/ULogViewer/blob/master/ULogViewer/ViewModels/DisplayableLogFilter.cs)

**Comment by acastruc** [comment]
@hamster620 ok Ive got something working but I am unclear about the current behaviour of combination mode. See my other post.

**Comment by acastruc** [comment]
@hamster620 I've got a branch I want to publish for inspection. How should I go about submitting?
When I try `git push -u origin <mybranchname>` I get 
```
remote: Permission to carina-studio/ULogViewer.git denied to acastruc.
fatal: unable to access 'https://github.com/carina-studio/ULogViewer.git/': The requested URL returned error: 403
```

**Comment by hamster620** [comment]
@acastruc You cannot push change to the repository directly. Please refer to [Creating a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) and create a pull request for merging chang to master branch.

To run and debug ULogViewer in VS Code, you need to edit ```launch.json``` and change the ```program``` attribute to correct framework:
![image](https://github.com/user-attachments/assets/174aecfb-0839-4915-9928-2974aee1a39c)
Currently it should be ```net8.0``` instead of ```net7.0```.

**Comment by acastruc** [comment]
@hamster620 understood. Done.

**Comment by acastruc** [comment]
@hamster620 can you give a rough timeline for when/if the exclude pattern will be supported in a new release?
Thanks

**Comment by hamster620** [comment]
> @hamster620 can you give a rough timeline for when/if the exclude pattern will be supported in a new release?
> Thanks

We can have a development release this week.

**Comment by hamster620** [comment]
The feature included in [5.0.0.921-dev](https://github.com/carina-studio/ULogViewer/releases/tag/5.0.0.921).
